### ISSUE TITLE
Unexpected DRR orientation

### ISSUE THREAD

**Issue by coolteemf** [arcadelab/deepdrr#99]
Hi
First of all, this plugin has been really helpful for my work, thanks a lot for developing it and making it open source !

Through testing, I have discovered a small issue with the orientation of the generated DRR.
When trying to relate 3D IJK coordinates to 2D pixel coordinates, there is a mismatch that can only be resolved by either transposing the image (X axis becomes Y axis) or the 2D coordinates which is equivalent.
This is obviously not a big issue as I can just transpose the image but I was curious as to why this is the case ?
Sample code:
```
volume = Volume.from_nifti(pth)
with Projector(volume,
               camera_intrinsics = camera_projection.intrinsic) as projector:
    proj = projector(camera_projection)
Y_slice = 250
# Points in IJK space
p_3D = [[179, Y_slice, 344], [209, Y_slice, 344], [209, Y_slice, 284], [179, Y_slice, 284]]
colors = ['red', 'orange', 'green', 'blue']
f, ax = plt.subplots()
# When plotting with matplotlib, images need to be transposed to have a horizontal first axis and a vertical second axis
ax.imshow(volume.data[:, Y_slice, ].T,
          origin='lower',
          cmap="gray")
for p, c in zip(p_3D, colors):
    ax.plot(p[0], p[2], '.', color=c)

f_, ax_ = plt.subplots()
ax_.imshow(proj.T, origin='lower', cmap="gray")
p2D = [camera_projection.index_from_world @ volume.world_from_ijk @ geo.point(p) for p in p_3D]
for p, c in zip(p2D, colors):
    ax_.plot(p[0], p[1], '.', color=c)
f_.suptitle("incorrect positions")

f__, ax__ = plt.subplots()
ax__.imshow(proj, origin='lower', cmap="gray")
for p, c in zip(p2D, colors):
    ax__.plot(p[0], p[1], '.', color=c)
f__.suptitle("correct positions")
```
Note: I have checked that the IJK points in the volume slice were correctly positioned with 3D Slicer, so I don't think this is an issue with the plotting.
![volume_slice](https://github.com/arcadelab/deepdrr/assets/67541941/d5e26a40-ec5e-4efd-a06a-cc0f1d2abd7c)
![incorrect_drr](https://github.com/arcadelab/deepdrr/assets/67541941/2e36e7fc-571b-41e5-9661-db52570cacf6)
![correct_drr](https://github.com/arcadelab/deepdrr/assets/67541941/22785e38-01d2-4df6-9162-3b7b3f181ad4)

**Comment by benjamindkilleen** [comment]
Can you try saving out the arrays as PNGs, either with cv2 or PIL (or deepdrr.utils.image_utils.save), rather than plotting with imshow? I notice the origin in a strange location, and I want to make sure matplotlib isn't the issue.

**Comment by coolteemf** [comment]
The location of the origin was to have images with up-down orientation matching the vertical, with the pixel (0,0) at the bottom of the image rather than the top.
Here are the arrays "proj" and "volume.data[:, Y_slice]" saved as is.
![projection](https://github.com/arcadelab/deepdrr/assets/67541941/bfb0bf91-6f1a-4773-b89b-c05a7bc25cfe)
![volume](https://github.com/arcadelab/deepdrr/assets/67541941/a4960bda-d0d2-41d9-8845-76ef73739862)

**Comment by benjamindkilleen** [comment]
This looks like matplotlib is responsible for part of the issue and cameraprojector for the other part. The X-axis is flipped due to the camera coordinate system we use, which aligns with the conventional pinhole camera model. Because the content in a C-arm system is between the image plane and the pinhole, as opposed to a conventional camera where it is on the other side of the pinhole, there is a compromise made between pointing the Z axis in the "right" direction and having X point in the expected direction. You can simply flip the image with `np.fliplr` to correct.

Separately, I highly recommend keeping the origin in the upper left, with x pointing right and y pointing down. It may seem counterintuitive, but it aligns with the image array index and is much more common in APIs like OpenCV.

**Comment by coolteemf** [comment]
Thanks for the explanation, it makes sense to me !

**Comment by coolteemf** [comment]
I followed your advice and kept the matplotlib plotting default.
I put the values in the volume in the rectangle shape to -1000.  at all depths and I then projected the modified volume:
![zeroed_volume](https://github.com/arcadelab/deepdrr/assets/67541941/2159a574-4484-4780-87ad-d8e600eb4a34)
![zeroed_drr](https://github.com/arcadelab/deepdrr/assets/67541941/19150c78-5a51-41ce-8c39-19092adc70f4)
I then computed as before the values of the 2D projection of the corners of the rectangle :
```
[Point2D([ 367.54343 1124.5035     1.     ]),
 Point2D([ 459.65396 1133.196      1.     ]),
 Point2D([477.03946 836.8651    1.     ]),
 Point2D([384.87198 828.54706   1.     ])]
```
This shows that the second coordinate of the 2D point correspond to the first axis in the image array, which points from the top left to the bottom left.
I checked that row number of the data array correspond to the up-down direction in the image.
So I think that the correct operation is to invert the two axes of the image rather than flipping the first dimension
![zeroed_drr_T](https://github.com/arcadelab/deepdrr/assets/67541941/350c1c43-db58-41cf-9658-9dbd289c503a)
